cloudferry_devlab/cloudferry_devlab/tests/testcases/test_resource_migration.py

```diff
@@ -108,37 +146,73 @@ def validate_network_name_in_port_lists(self, src_ports, dst_ports):
+            value = dict(token.split('=') for token in attr_list)
+            return value[attribute]
+        except TypeError:
+            return None
+
+    def test_migrate_keystone_users_name(self):
+        """Validate users were migrated with correct name."""
         src_users = self.filter_users()
         dst_users = self.dst_cloud.keystoneclient.users.list()
+        self.validate_tenant_resource_param_in_dst(src_users,
+                                                   dst_users,
+                                                   resource_name='user',
+                                                   parameter='name')
 
-        for param in ['name', 'email']:
-            self.validate_resource_parameter_in_dst(src_users, dst_users,
-                                                    resource_name='user',
-                                                    parameter=param)
-
-    @attr(migrated_tenant=['admin', 'tenant1', 'tenant2'])
+    def test_migrate_keystone_users_email(self):
+        """Validate users were migrated with correct email."""
+        src_users = self.filter_users()
+        dst_users = self.dst_cloud.keystoneclient.users.list()
+        src_users = [user for user in src_users
+                     if getattr(user, "name") !=
+                     config.case_sensitivity_test_user]
+        self.validate_tenant_resource_param_in_dst(src_users,
+                                                   dst_users,
+                                                   resource_name='user',
+                                                   parameter='email')
+
+    @attr(migrated_tenant=['admin', 'tenant1', 'tenant2', 'tenant6'])
     def test_migrate_keystone_user_tenant_roles(self):
         """Validate user's tenant roles were migrated with correct name."""
         src_users = self.filter_users()
-        src_user_names = [user.name for user in src_users]
         dst_users = self.dst_cloud.keystoneclient.users.list()
+        tenant = self.get_nosetest_cmd_attribute_val('migrated_tenant')
         least_user_match = False
         for dst_user in dst_users:
-            if dst_user.name not in src_user_names:
+            src_user = None
+            for user in src_users:
+                if user.name.lower() == dst_user.name.lower():
+                    src_user = user
+                    break
+            if src_user is None:
                 continue
             least_user_match = True
-            src_user_tnt_roles = self.src_cloud.get_user_tenant_roles(dst_user)
-            dst_user_tnt_roles = self.dst_cloud.get_user_tenant_roles(dst_user)
+            src_user_tnt_roles = self.src_cloud.get_user_tenant_roles(
+                                     src_user, tenant)
+            dst_user_tnt_roles = self.dst_cloud.get_user_tenant_roles(
+                                     dst_user, tenant)
+            if len(src_user_tnt_roles) == 0 and len(dst_user_tnt_roles) == 0:
```

Comment:
This check looks unnecessary (unless there's some invalid behavior added to `get_user_tenant_roles` method). Roles should be verified by `validate_resource_parameter_in_dst`.